lib/pages/RegistrationForm.dart

```diff
@@ -0,0 +1,215 @@
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:math';
+import 'package:convert/convert.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:cryptography/cryptography.dart';
+import 'package:hive/hive.dart';
+
+import 'HomePages.dart';
+import 'SplashScreen.dart';
+
+Future<Map> _calculateEncryptionValues(String pin) async {
+  HashMap values = HashMap();
+
+  // this should give us 256 bits? someone check my math
+  final SecretKey randKey = SecretKey(
+      List<int>.generate(32, (index) => Random.secure().nextInt(255)));
+  final List<int> randKeyBytes = await randKey.extractBytes();
+  values.putIfAbsent("randKey", () => randKeyBytes);
```

Comment:
This is why I dislike Flutter. Like why is this implementation a thing? Why would you have a function as the second argument here? It looks disgusting.

It's totally fine but I just wanted to vent.

Reply:
Hmm you can also do values["randKey"] = randKeyBytes, but I did putIfAbsent here for some reason. I'll change this segment to be more consistent with the style of the rest of the project